hw3-todo-cli/todocli.js

```diff
@@ -0,0 +1,199 @@
+}
+
+
+// a switch statement/function to triage the command the user type, and call the appropriate function
+
+rl.on('line', (command) => {
+    switch (command[0]){
+        case 'v':
+            console.log("\n \n 👉 View 👈\n \n")
+            viewItem() 
+            break;
+        case 'c':
+            console.log("\n \n 👉 Mark Complete 👈\n \n ")
+            completeItems(command.slice(1)) // might have to turn this into slice later, in case we get to double digits
+            break;
+        case 'n':
+            console.log("\n \n 👉 Add New List Item 👈\n \n ")
+            newItem()
+            break;
+        case 'd':
+            console.log("\n \n 👉 Delete a List Item 👈\n \n ")
+            deleteItem(command.slice(1)) // might have to turn this into slice later, in case we get to double digits
+            break;
+        case 'q':
+            console.log("\n \n 👉 Quit 👈\n \n ")
+            console.log("\n \n Finished already? Great work! See you soon😃\n \n")
+            rl.close()
+            break;
+        case 's':
+            console.log("\n \n 👉 Save List 👈\n \n ")
+            saveToDo()
+            break;
+        default:
+            console.log("Hmm... that didn't work. Did ya wanna try that again? \n \n(v) View * (n) New * (cX) Complete * (dX) Delete * (q) Quit")
+            //r1.prompt()
+    }
+});
+
+    //displays the list in a nice, easy to read format. 
+    // Delays a half a sec from the command being received 
+function viewItem(){
+    if (tasks.length > 0){
+        setTimeout(function() {
+        console.log("📝 To Do List\n")
+        for (let i = 0; i < tasks.length; i++) {
+            console.log(i, tasks[i].join(" "))  
+        }
+        commandList()
+        }, 500)
+        } else {
+        console.log("Your To Do list is empty, you can add an item but pressing 'n'")
+    }
+}
+
+// adds a new item to the list, pushing it to the array that is processed in viewItem()
+
+function newItem(){
+    rl.question("What new item would you like to add to the list? \n", (newitem) => {
+
+        tasks.push(["[ ]",newitem])
+
+        setTimeout(function() {
+            console.log(`\n \n ✅${newitem} has been added to your to-do list\n \n`)
+        commandList()
+        }   , 500);
+        })  
+    }
+
+// marks an item on the list as complete by the index. Error logging for
+// if list is empty, and for if the index is out of range or can't be interpreted.
+// each provides a helpful suggestion
+   
+function completeItems(checkOff){
+    if (tasks.length == 0){
+        console.log("Whoops! Your To Do list is empty, you can add an item but pressing 'n'")
+    } else {
+        try{
+            console.log(`Marking: '${tasks[checkOff][1]}' complete...\n`)
+            tasks[checkOff][0] = "[✓]"
+
+            //setting the timeout just because I think it looks nice 
+
+            setTimeout(function() {
```

Comment:
nice use of `setTimeout`. But just a note on the user experience side, it feels like there's a glitch. More suitable use case of this would be when you are dealing with a race condition.